src/azul/service/async_manifest_service.py

```diff
@@ -63,40 +78,86 @@ class AsyncManifestService(AbstractService):
     """
     step_function_helper = StepFunctionHelper()
 
+    step_function_execution_retries = 2
+
     def __init__(self, state_machine_name):
         self.state_machine_name = state_machine_name
 
     def start_generation(self, input: JSON) -> Token:
-        execution_id = str(uuid.uuid4())
-        self.step_function_helper.start_execution(self.state_machine_name,
-                                                  execution_id,
-                                                  execution_input=input)
-        return Token(execution_id=execution_id,
-                     request_index=0,
-                     wait_time=self._get_next_wait_time(0))
-
-    def inspect_generation(self, token) -> Union[Token, JSON]:
+        execution_id = input['object_key'].rpartition('/')[2]
+        execution_id = execution_id.partition('.')[0]
+        token = Token(execution_id=execution_id,
+                      request_index=0,
+                      wait_time=self._get_next_wait_time(0))
+        return self._start_generation(input, token)
+
+    def _start_generation(self,
```

Comment:
Try nested functions for this and the retry bit. If there is shared state between them it will simplify their signature, because that state will be in scope rather than having to be passed in.